agent/src/main/java/com/teamscale/jacoco/agent/git_properties/GitPropertiesLocatingTransformer.java

```diff
@@ -0,0 +1,69 @@
+package com.teamscale.jacoco.agent.git_properties;
+
+import com.teamscale.jacoco.agent.util.LoggingUtils;
+import org.conqat.lib.commons.string.StringUtils;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.lang.instrument.ClassFileTransformer;
+import java.net.URL;
+import java.security.CodeSource;
+import java.security.ProtectionDomain;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+/**
+ * {@link ClassFileTransformer} that doesn't change the loaded classes but searches their corresponding Jar/War/Ear/...
+ * files for a git.properties file.
+ */
+public class GitPropertiesLocatingTransformer implements ClassFileTransformer {
+
+	private final Logger logger = LoggingUtils.getLogger(this);
+	private final Set<String> seenJars = new ConcurrentSkipListSet<>();
+	private final GitPropertiesLocator locator;
+
+	public GitPropertiesLocatingTransformer(GitPropertiesLocator locator) {
+		this.locator = locator;
+	}
+
+	@Override
+	public byte[] transform(ClassLoader classLoader, String className, Class<?> aClass,
+							ProtectionDomain protectionDomain, byte[] classFileContent) {
+
+		if (protectionDomain == null) {
+			// happens for e.g. java.lang. We can ignore these classes
+			return null;
+		}
+
+		try {
+			CodeSource codeSource = protectionDomain.getCodeSource();
+			if (codeSource == null) {
+				// unknown when this can happen, but we need to be safe and there's
+				// nothing else we can do here
```

Comment:
Doesn't really matter, but I guess this can happen when you generate bytecode programmatically e.g. with the asm lib or via lombok